### PR TITLE
fix: Feature/iframe dynamic visibility

### LINKED ISSUE #27991: [Feature]: Add support for isVisible property in the iframe widget
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Iframe widget cannot be hidden or shown dynamically, making it a blocker for cases like auto-height in containers. This feels like an inconsistency.

### Why should this be worked on?

This will help developers create a better experience for their flows where they want to show iframe dynamically

### PR DESCRIPTION
**Pull request by danceAndJive**
#### PR fixes following issue(s)
Fixes #27991 
>
>
#### Media
275673732-007a8684-69b0-4c31-97eb-c1aa7e373095

>
>
#### Type of change
- New feature (non-breaking change which adds functionality)
>
>
>
- [x] Manual
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by Aishwarya-U-R — COMMENTED** [review]
Yml file changes are ok.

**Comment by dilippitchika** [comment]
Hey @danceAndJive thanks for the contribution, there is one change which we need to you make to adhere with our quality standards. In our edit mode, the widget is not fully hidden but it's visible with less transparency, while in the deployed mode it's fully hidden. 

Please check any other widget like checkbox to know how this works - https://www.loom.com/share/989efc440f7749049b8d2c7d62d116bc

Please let me know if you can fix this, thanks and appreciate your contribution

File: app/client/package.json [hunk 1]
```diff
@@ -289,6 +288,7 @@
     "@typescript-eslint/parser": "^6.7.4",
     "babel-plugin-lodash": "^3.3.4",
     "babel-plugin-module-resolver": "^4.1.0",
+    "canvas": "^2.11.2",
```
Comment (KelvinOm) [inline app/client/package.json:291]: @riodeuno Could you merge your [PR](https://github.com/appsmithorg/appsmith/pull/28273) to avoid conflicts?

**Comment by riodeuno** [comment]
@KelvinOm One test is failing. I'll merge it today.

**Comment by somangshu** [comment]
Looks like there was a dirty back merge here @riodeuno @KelvinOm 

@dilippitchika can you check the DP again?

**Comment by danceAndJive** [comment]
Hi @dilippitchika, thank you for the feedback. I have undone the code change I had made in iframe widget component index.tsx, and now it seems to be working in a similar fashion to the checkbox widget. Please let me know if any further changes are required. Thanks.

**Comment by somangshu** [comment]
@KelvinOm i dont think this is a clean merge. Please verify once on your end. 
Approve code changes as well please

**Comment by KelvinOm** [comment]
@danceAndJive Please clean up code in your PR leave only what is related to the issue being solved.

**Comment by danceAndJive** [comment]
> @danceAndJive Please clean up code in your PR leave only what is related to the issue being solved.

Hi @KelvinOm, sorry I'm really new to this. 

I rebased before committing my changes to my branch. Could that be the reason why codes changes from other people are introduced to my branch? 

I'm not sure how to clean up other code since I did not change them myself. Would it be better for me to create a new branch, fix the issue and make another PR without rebasing?

Thanks.

**Comment by somangshu** [comment]
@danceAndJive that is correct. You need to instead sync your folk (back merge our release branch into you forked branch). [check out this article for more details](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork)

You can clean up the current branch by reverting commits or create a new branch if the former sounds complex.

**Comment by KelvinOm** [comment]
> I'm not sure how to clean up other code since I did not change them myself. Would it be better for me to create a new branch, fix the issue and make another PR without rebasing?

@danceAndJive Create a new branch(PR) and apply only the necessary changes to it. This should work. Let me know if you need any help.

**Comment by danceAndJive** [comment]
Hi @somangshu and @KelvinOm, thanks for your suggestions! I have created a new PR #28397 with only necessary changes. Please let me know should there be any issues.

**Comment by somangshu** [comment]
Thanks @danceAndJive, LGTM, closing this in favour of the new one

**Pull request closed without merging**